servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/MessageReceiver.java

```diff
@@ -695,9 +695,7 @@ private void ensurePeekLockReceiveMode() {
                 for(UUID lockToken : lockTokens) {
                     if (lockTimeIterator.hasNext()) {
                         Instant lockedUntilUtc = lockTimeIterator.next();
-                        if (this.requestResponseLockTokensToLockTimesMap.containsKey(lockToken)) {
-                            this.requestResponseLockTokensToLockTimesMap.put(lockToken, lockedUntilUtc);
-                        }
+                        this.requestResponseLockTokensToLockTimesMap.computeIfPresent(lockToken, (k, v) -> lockedUntilUtc);
```

Comment:
[SpotBugs-P3]

Sequence of calls to java.util.concurrent.ConcurrentHashMap may not be atomic in com.microsoft.azure.servicebus.MessageReceiver.lambda$renewMessageLockBatchAsync$20(UUID[], Collection)